### DISCUSSION TITLE
Per-judgement hitsound skinning

### DISCUSSION THREAD

**Discussion by aryoadhi** [ppy/osu#23688] [Ideas]
This quirk comes from mobile rhythm games (especially VSRGs), where notes may play different sound depending on how accurate the players are, or whether the lanes are empty or not. This includes, but not limited to: perfect, great, good, and bad hits. This is great to accomodate these players where they're mostly using these hitsound scheme, so they can adapt more easily to osu!.

This proposal mostly seen its application to osu!mania ruleset, but I think this can be applied too to other modes like osu!taiko and osu!. osu!catch is not feasible in this idea due to you're either hitting or misisng the fruits completely.

The file names can be like this:
```
normal-empty.wav
normal-hitnormal-100.wav
normal-hitnormal-geki.wav
normal-hitnormal-katu.wav
```

The file structure idea can be better to incorporate 3 modes without making the skin too bloated with multiple files. Maybe there should be a toggle too in options whether users want to use per-judgement hitsounds, since it should be an optional settings, not forced.

**Comment by SirHonkkk** [comment]
i like this but they removed geki and katu from the game tho. i would like to see them come back

**Comment by Fivoka** [comment]
No toggle needed, just let the game use the hitsound of a 300 if it can't find hitsounds for 100s and 50s
